api/control/v1alpha1/types.go

```diff
@@ -490,8 +490,24 @@ type ReturnedState struct {
 type PropagationData struct {
 	// `lastReturnedUpdateTimestamp` is the time of the last update to any
 	// of the returned object state in the core.
-	// Before the first such update, this is the zero value of `time.Time`.
+	// Before the first such update, this holds the zero value of `time.Time`.
 	LastReturnedUpdateTimestamp metav1.Time `json:"lastReturnedUpdateTimestamp"`
+
+	// `lastGeneration` is that last `ObjectMeta.Generation` from the WDS that
+	// propagated to the WEC. This is not to imply that it was successfully applied there;
+	// for that, see `lastGenerationIsApplied`.
+	// Zero means that none has yet propagated there.
+	LastGeneration int64 `json:"lastGeneration"`
+
+	// `lastGenerationIsApplied` indicates whether `lastGeneration` has been successfully
+	// applied in the WEC. This is `false` while no attempt has yet been made.
```

Comment:
I think the 2nd part of the sentence is redundant and may introduce confusion. This bit should be `true` only if `LastGeneration` is applied. That's it. Its previous state or the reason it is/was switched to false is irrelevant, in my opinion.

Reply:
To mind mind also, the second sentence is redundant. I included it for clarity --- I think that _some_ other people will think that the first sentence does not cover this case.

Reply:
Since it confused @nirrozenbaum, I have removed it.